src/gui/torrentcontentmodel.cpp

```diff
@@ -158,30 +160,24 @@ namespace
         return (!testIcon1.isNull() || !testIcon2.isNull());
     }
 
-    class MimeFileIconProvider : public UnifiedFileIconProvider
+    class MimeFileIconProvider final : public UnifiedFileIconProvider
     {
         using QFileIconProvider::icon;
 
         QIcon icon(const QFileInfo &info) const override
         {
-            const QMimeType mimeType = m_db.mimeTypeForFile(info, QMimeDatabase::MatchExtension);
-            QIcon res = QIcon::fromTheme(mimeType.iconName());
-            if (!res.isNull())
-            {
-                return res;
-            }
+            const QMimeType mimeType = QMimeDatabase().mimeTypeForFile(info, QMimeDatabase::MatchExtension);
```

Comment:
Are you sure constructing QMimeDatabase is cheap?

Reply:
https://doc.qt.io/qt-5/qmimedatabase.html#QMimeDatabase-1
>It is perfectly OK to create an instance of QMimeDatabase every time you need to perform a lookup.